src/dstack/_internal/core/backends/cloudrift/compute.py

```diff
@@ -0,0 +1,131 @@
+from typing import Dict, List, Optional
+
+from dstack._internal.core.backends.base.backend import Compute
+from dstack._internal.core.backends.base.compute import (
+    ComputeWithCreateInstanceSupport,
+    get_shim_commands,
+)
+from dstack._internal.core.backends.base.offers import get_catalog_offers
+from dstack._internal.core.backends.cloudrift.api_client import RiftClient
+from dstack._internal.core.backends.cloudrift.models import CloudRiftConfig
+from dstack._internal.core.errors import ComputeError
+from dstack._internal.core.models.backends.base import BackendType
+from dstack._internal.core.models.instances import (
+    InstanceAvailability,
+    InstanceConfiguration,
+    InstanceOffer,
+    InstanceOfferWithAvailability,
+)
+from dstack._internal.core.models.runs import JobProvisioningData, Requirements
+from dstack._internal.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class CloudRiftCompute(
+    ComputeWithCreateInstanceSupport,
+    Compute,
+):
+    def __init__(self, config: CloudRiftConfig):
+        super().__init__()
+        self.config = config
+        self.client = RiftClient(self.config.creds.api_key)
+
+    def get_offers(
+        self, requirements: Optional[Requirements] = None
+    ) -> List[InstanceOfferWithAvailability]:
+        offers = get_catalog_offers(
+            backend=BackendType.CLOUDRIFT,
+            locations=self.config.regions or None,
+            requirements=requirements,
+        )
+
+        offers_with_availabilities = self._get_offers_with_availability(offers)
+        return offers_with_availabilities
+
+    def _get_offers_with_availability(
+        self, offers: List[InstanceOffer]
+    ) -> List[InstanceOfferWithAvailability]:
+        instance_types_with_availabilities: List[Dict] = self.client.get_instance_types()
+
+        region_availabilities = {}
+        for instance_type in instance_types_with_availabilities:
+            for variant in instance_type["variants"]:
+                for dc, count in variant["available_nodes_per_dc"].items():
+                    if count > 0:
+                        key = (variant["name"], dc)
+                        region_availabilities[key] = InstanceAvailability.AVAILABLE
+
+        availability_offers = []
+        for offer in offers:
+            key = (offer.instance.name, offer.region)
+            availability = region_availabilities.get(key, InstanceAvailability.NOT_AVAILABLE)
+            availability_offers.append(
+                InstanceOfferWithAvailability(**offer.dict(), availability=availability)
+            )
+            logger.debug(
+                f"Offer {offer.instance.name} in region {offer.region} has availability: {availability}"
+            )
+
+        return availability_offers
+
+    def create_instance(
+        self,
+        instance_offer: InstanceOfferWithAvailability,
+        instance_config: InstanceConfiguration,
+    ) -> JobProvisioningData:
```

Comment:
I'm getting errors because of a missing parameter in the signature. This is a new parameter, so I assume it wasn't there when you started working on the backend.

```suggestion
    def create_instance(
        self,
        instance_offer: InstanceOfferWithAvailability,
        instance_config: InstanceConfiguration,
        placement_group: Optional[PlacementGroup],
    ) -> JobProvisioningData:
```

Will also need to import `PlacementGroup`